src/lib/slider/slider.ts

```diff
@@ -226,6 +226,12 @@ export class MdSlider implements ControlValueAccessor {
   set vertical(value: any) { this._vertical = coerceBooleanProperty(value); }
   private _vertical = false;
 
+  /** The value to be used for display purposes. */
+  get displayValue(): string|number {
+    // Skip adding the decimal part if the number is whole.
+    return this.value % 1 === 0 ? this.value : this.value.toFixed(1);
```

Comment:
That's what I was thinking initially, however anything longer than 3 characters doesn't look right with the current setup.